### PR TITLE
Add telemetry

### PR DESCRIPTION
**Pull request by arroyc**
adding user-agent for tracking azure storage plugin telemetry

### CODE REVIEW

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java [hunk 1]
```diff
@@ -141,7 +146,9 @@ private static CloudBlobContainer getBlobContainerReference(StorageAccountInfo s
 	boolean cntExists = container.exists();
 
 	if (createCnt && !cntExists) {
-	    container.createIfNotExists();
+            String userInfo = container.getUri().toURL().getUserInfo();
+            //container.uploadMetadata(null, null, Utils.injectUserAgent(userInfo));
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: looks like you forgot to remove the comment

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java [hunk 2]
```diff
@@ -374,11 +385,13 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
...
      */
     private static void deleteContents(CloudBlobContainer container)
-	    throws StorageException, URISyntaxException {
+	    throws StorageException, URISyntaxException, MalformedURLException {
 
 	for (ListBlobItem blobItem : container.listBlobs()) {
 	    if (blobItem instanceof CloudBlob) {
-		((CloudBlob) blobItem).delete();
+                String info = blobItem.getUri().toURL().getUserInfo();
+                //((CloudBlob) blobItem).uploadProperties(null, null, Utils.injectUserAgent(info));
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: clean comment

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java [hunk 3]
```diff
@@ -393,11 +406,13 @@ private static void deleteContents(CloudBlobContainer container)
...
      */
     private static void deleteContents(CloudBlobDirectory cloudBlobDirectory)
-	    throws StorageException, URISyntaxException {
+	    throws StorageException, URISyntaxException, MalformedURLException {
 
 	for (ListBlobItem blobItem : cloudBlobDirectory.listBlobs()) {
 	    if (blobItem instanceof CloudBlob) {
-		((CloudBlob) blobItem).delete();
+                String info = blobItem.getUri().toURL().getUserInfo();
+                //((CloudBlob) blobItem).uploadProperties(null, null, Utils.injectUserAgent(info));
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: one more :)

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java [hunk 4]
```diff
@@ -133,5 +135,41 @@ public static String getMD5(String plainText) {
...
+            version = "local";
+        }
+        if (instanceId == null) {
+            instanceId = "local";
+        }
+
+        return Constants.PLUGIN_NAME + "/" + version + "/" + instanceId;
+    }
+    
+    public static OperationContext injectUserAgent(String userInfo) {
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java]: I would hide the required ugliness for getting the userInfo inside the injectUserAgent.
Overload the method with different argument types for all the SDK objects like CloudBlob, CloudBlobContainer and in each of these fetch the userInfo and call the method with the string argument.
In this way you won't have to have that extra line in each call

Comment (arroyc) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java]: agreed partially and changed the code accordingly ...

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java [hunk 5]
```diff
@@ -133,5 +139,39 @@ public static String getMD5(String plainText) {
     public static String getWorkDirectory() {
         return Jenkins.getInstance().root.getAbsolutePath();
     }
+    
+    public static OperationContext updateUserAgent(URL getInfo) throws IOException {
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java]: why do you need the 'getInfo' argument?

Comment (arroyc) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java]: yes I don't need it anymore ... :) forgot cleaning !!

**Pull request merged**